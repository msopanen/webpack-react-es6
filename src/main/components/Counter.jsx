import React from "react";
import { render } from "react-dom";
import PropTypes from "prop-types";
import { connect } from "react-redux";

class Counter extends React.Component {
    
    static propTypes = {
        counter: PropTypes.number,
        onIncrement: PropTypes.func,
        onDecrement: PropTypes.func,
    };

    render() {

        const { counter, onDecrement, onIncrement } = this.props;

        return (
            <div>
                <div>{counter}</div>
                <button onClick={onDecrement}>-</button>
                <button onClick={onIncrement}>+</button>
            </div>
        );
    }
};

const mapStateToProps = (state) => {
    return state;
};

const mapDispatchToProps = (dispatch) => {
    return {
        onIncrement: () => dispatch({ type: "INCREMENT" }),
        onDecrement: () => dispatch({ type: "DECREMENT" }),
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(Counter);