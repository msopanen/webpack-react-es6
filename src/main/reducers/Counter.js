export default (state, action) => {
    switch (action.type) {
        case 'INCREMENT':
            return { ...state, counter: state.counter + 10 };
        case 'DECREMENT':
            return { ...state, counter: state.counter - 1 };
        default:
            return state;
    }
};