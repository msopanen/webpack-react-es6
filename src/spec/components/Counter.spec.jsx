
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import Counter from "../../main/components/Counter";

const middlewares = [];
const mockStore = configureMockStore(middlewares);

it("renders initial state 0 properly",() => {

    const getState = { counter: 0 }; // initial state of the store

    const store = mockStore(getState, []);
    const tree = renderer.create(
        <Counter store={store} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

/*it("renders properly", (done) => {

    const getState = { counter: 0 }; // initial state of the store
    const increment = { type: 'INCREMENT' };
    const expectedActions = [increment];

    const store = mockStore(getState, expectedActions, done);
    const tree = renderer.create(
        <Counter store={store} />
    ).toJSON();
    expect(tree).toMatchSnapshot();

    store.dispatch(increment);
});*/