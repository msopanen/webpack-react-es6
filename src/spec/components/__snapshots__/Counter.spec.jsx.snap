// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders initial state 0 properly 1`] = `
<div>
  <div>
    0
  </div>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`renders properly 1`] = `
<div>
  <div>
    0
  </div>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
